vcd/resource_vcd_catalog_media.go

```diff
@@ -196,11 +196,12 @@ func resourceVcdMediaCreate(ctx context.Context, d *schema.ResourceData, meta in
 		for {
 			progress, err := task.GetTaskProgress()
 			if err != nil {
-				log.Printf("vCD Error importing new catalog item: %s", err)
-				return diag.Errorf("vCD Error importing new catalog item: %s", err)
+				log.Printf("VCD Error importing new catalog item: %s", err)
+				return diag.Errorf("VCD Error importing new catalog item: %s", err)
 			}
-			logForScreen("vcd_catalog_media", fmt.Sprintf("vcd_catalog_media."+mediaName+": vCD import catalog item progress "+progress+"%%\n"))
-			if progress == "100" {
+			logForScreen("vcd_catalog_media", fmt.Sprintf("vcd_catalog_media.%s: VCD import catalog item progress %s%%\n", mediaName, progress))
```

Comment:
This loop exits in two cases:
* when the task status is "error" (returned from `task.GetTaskProgress') 
* when the status is "success".

What about the other status values? Probably we should considered `"aborted"` in this context, as other statuses refer to transient operations ("queued", "running", "preRunning")

Reply:
We could probably factor in aborted to catch the error if the user presses "cancel" in UI.

Reply:
We could do more than caching the error: an interruption at this stage will leave the media item empty in the catalog, preventing a re-run of the HCL script (the item is already in the catalog)

Reply:
But maybe this is beyond the scope of this change

Reply:
Indeed! I have added `aborted` to the exit condition of the loop.

I checked that `task.WaitTaskCompletion()` (after the modified loop) finishes without error if the task is aborted, but since it was also the original behavior, I'm not changing anything there.

Reply:
Ah, I didn't see the other replies to this thread 🙂 

I thought the same @dataclouder (adding a condition that if it's aborted it should fail), but I'm also concerned about changing too much.

Reply:
Finally added the check of `aborted` to throw an error in that case. Test passes now.